src/recon_test/test_consistency_with_root.cxx

```diff
@@ -0,0 +1,547 @@
+  test_results_nonTOF = std::vector<bool> (num_tests, true);
+  test_results_TOF = std::vector<bool> (num_tests, true);
+
+  cerr << "Testing the view offset consistency between GATE/ROOT and STIR. \n";
+  for (int i = 1; i <= num_test; ++i)
+  {
+    test_index = i;
+    cerr << "\nTesting dataset " << std::to_string(test_index) << "...\n";
+
+    setup();
+    process_list_data();
+
+    cerr << "\nResults for dataset: " << std::to_string(test_index) << std::endl;
+    post_processing_nonTOF();
+    post_processing_TOF();
+  }
+
+  // Print results for easy readability
+  cerr << "\nTest Results\n"
+          "------------------------------------------\n"
+          "\tTest Index\t|\tnonTOF\t|\tTOF\n"
+          "------------------------------------------\n";
+  for (int j = 0; j < num_test; ++j)
+    cerr << "\t\t" << std::to_string(j + 1) << "\t\t|\t"
+         << ((test_results_nonTOF[j]) ? "Pass" : "Fail") << "\t|\t"
+         << ((test_results_TOF[j]) ? "Pass" : "Fail") << std::endl;
+}
+
+void
+ROOTConsistencyTests::
+setup()
+{
+  {
+    // Create the point source (discretised_density_sptr) with GenerateImage from parameter file
+    // GenerateImage requires `const char* const par_filename`.
+    GenerateImage image_gen_application(get_generate_image_par_filename().c_str());
+    image_gen_application.compute();
+    discretised_density_sptr = image_gen_application.get_output_sptr();
+  }
+
+  // Needs to be cast to VoxelsOnCartesianGrid to be able to calculate the centre of gravity
+  const VoxelsOnCartesianGrid<float> &discretised_cartesian_grid =
+      dynamic_cast<VoxelsOnCartesianGrid<float> &>(*discretised_density_sptr);
+
+  // Find the center of mass and grid spacing of the original data
+  original_coords = find_centre_of_gravity_in_mm(discretised_cartesian_grid);
+  grid_spacing = discretised_cartesian_grid.get_grid_spacing();
+
+  // For post_processing and debugging, create a list of the closest LOR coords
+  nonTOF_closest_voxels_list.empty();
+  max_coords_for_each_event.empty();
+
+  // The number of LORs with closes approach greater than the threshold.
+  num_failed_nonTOF_lor_events = 0;
+  num_failed_TOF_lor_events = 0;
+  num_events_tested = 0;
+
+  // Failure conditioner and recording
+  nonTOF_distance_threshold = 1.5 * norm(grid_spacing); // Using norm(grid_spacing) as a nonTOF_distance_threshold
+  TOF_distance_threshold = 1.5 * norm(grid_spacing); // Using norm(grid_spacing) as a tof_distance_threshold
+}
+
+void
+ROOTConsistencyTests::process_list_data()
+{
+  // Configure the list mode reader
+  shared_ptr<CListModeData> lm_data_sptr(read_from_file<CListModeData>(get_root_header_filename()));
+  shared_ptr<ProjMatrixByBin> proj_matrix_sptr(new ProjMatrixByBinUsingRayTracing());
+  proj_matrix_sptr.get()->set_up(lm_data_sptr->get_proj_data_info_sptr(),
+                                 discretised_density_sptr);
+
+  // TOF requirements
+  proj_matrix_sptr->enable_tof(lm_data_sptr->get_proj_data_info_sptr());
```

Comment:
Build error
```
/home/runner/work/STIR/STIR/src/include/stir/recon_buildblock/ProjMatrixByBin.h:246:8: note: declared private here
  246 |   void enable_tof(const shared_ptr<const ProjDataInfo>& proj_data_info_sptr,const bool v = true);
      |        ^~~~~~~~~~
```

Locally I have moved `enable_tof` to a public method. Should we do this in the TOF branch?

Reply:
In the latest change, the `enable_tof()` call was removed.